### PR TITLE
added callbacks to versionAll

### PR DESCRIPTION
**Pull request by arturoolvera**
Modified versionAll to allow for callback functions so we can see which servers are not responding/timing out.

### CODE REVIEW

File: src/memjs/memjs.ts [hunk 1]
```diff
@@ -909,14 +914,22 @@ class Client<Value = MaybeBuffer, Extras = MaybeBuffer> {
    * in the backend pool, errors if any one of them has an
    * error
    */
-  async versionAll(): Promise<{
+  async versionAll(
+    triedCallback?: (response: string) => void,
```
Comment (leon-root) [inline src/memjs/memjs.ts]: Maybe write down how to use these callback in the comments.

File: src/memjs/memjs.ts [hunk 2]
```diff
@@ -909,14 +914,22 @@ class Client<Value = MaybeBuffer, Extras = MaybeBuffer> {
...
+    resultCallback?: (response: string) => void
+  ): Promise<{
     values: Record<string, Value | null>;
   }> {
     const versionObjects = await Promise.all(
       this.serverKeys.map((serverKey) => {
+        if (triedCallback !== undefined) {
+          triedCallback(serverKey);
+        }
         const server = this.serverKeyToServer(serverKey);
```
Comment (leon-root) [inline src/memjs/memjs.ts:928]: Did you compare `serverKey` with `server` for callback? which one is more descriptive?

Comment (arturoolvera) [reply src/memjs/memjs.ts:928]: synced offline -- looks like there's no extra info that we get by passing `server` instead of `serverKey`

**Comment by blackmad** [comment]
is this needed because this function doesn't return when hosts are down?

**Comment by arturoolvera** [comment]
> is this needed because this function doesn't return when hosts are down?

yes. the call to versionAll is called with a timeout and this lets us track which hosts aren't returning.
Associated pr [here](https://github.com/makenotion/notion-next/pull/31432)

**Comment by blackmad** [comment]
why not just add a timeout param to this call?

**Pull request merged**